src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs

```diff
@@ -372,7 +372,13 @@ internal static void ResetThreadPoolThread(Thread currentThread)
         [System.Diagnostics.Conditional("DEBUG")]
         internal static void CheckThreadPoolAndContextsAreDefault()
         {
-            Debug.Assert(Thread.CurrentThread.IsThreadPoolThread);
+            if (Thread.IsThreadStartSupported)
+            {
+#pragma warning disable CS0162 // Unreachable code detected. Thread.IsThreadStartSupported is constant false in runtimes where we can't create threads.
+                Debug.Assert(Thread.CurrentThread.IsThreadPoolThread);
+#pragma warning restore CS0162
+            }
```

Comment:
If this were instead written as:
```
Debug.Assert(!Thread.IsThreadStartSupported || Thread.CurrentThread.IsThreadPoolThread);
```
would it still need the warning suppression?

Also, this use seems less about thread start and more about whether the thread pool has multiple threads. Maybe it should have its own const on thread pool?

Reply:
Yep that works.

> this use seems less about thread start and more about whether the thread pool has multiple threads.

Yeah though isn't that kinda related? Without thread start you can't have multiple threads :)

Reply:
I added a comment to explain this a bit more.